cli/src/pcluster/templates/cdk_builder.py

```diff
@@ -12,15 +12,151 @@
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                # TODO: Fix the bug using data property or uri with core.Fn.sub
+                uri="s3://test-slurm/official_ami_build_component.yml",
+            )
+        else:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster custom AMI",
+                change_description="First version",
+                platform="Linux",
+                # TODO: Fix the bug using data property or uri with core.Fn.sub
+                uri="s3://test-slurm/custom_ami_build_component.yml",
+            )
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="PCluster-2-10-1",
+            version="0.0.1",
+            parent_image=core.Fn.sub(
+                "arn:${AWS::Partition}:imagebuilder:${AWS::Region}:aws:image/amazon-linux-2-x86/x.x.x"
+                if not imagebuild.config.build_config.parent_image
+                else imagebuild.config.build_config.parent_image
+            ),
+            components=[
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                )
+            ],
+            block_device_mappings=[
+                imagebuilder.CfnImageRecipe.InstanceBlockDeviceMappingProperty(
+                    device_name="/dev/xvda",
+                    ebs=imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+                        delete_on_termination=imagebuild.config.image_config.root_volume.delete_on_termination,
+                        volume_size=imagebuild.config.image_config.root_volume.size,
+                        volume_type="gp2",
+                    )
+                    if imagebuild.config.image_config.root_volume
+                    else imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+                        delete_on_termination=True, volume_size=25, volume_type="gp2"
+                    ),
+                )
+            ],
+            tags={"ParallelClusterVersion": "2.10.1"},
+        )
+
+        # Image
+        imagebuilder.CfnImage(
+            self,
+            id="PClusterImage",
+            image_recipe_arn=core.Fn.ref("PClusterImageRecipe"),
+            infrastructure_configuration_arn=core.Fn.ref("PClusterImageInfrastructureConfiguration"),
+        )
+
+        # AWS Systems Manager
+        ssm.CfnParameter(
```

Comment:
Why ssm.CfnParameter and not core.CfnParameter ?

Reply:
we defined a SSM rather than cfn parameter here, so we are able to store the AMI ID in a SSM parameter store and retrieve this from SSM parameter store.